### ISSUE TITLE
Updated file content is not respected when running before explicitly saving

### ISSUE THREAD

**Issue by izzyreal** [scalameta/metals#5337]
### Describe the bug

Using Metals v0.11.12 - Aluminium with VSCodium Version: 1.79.1

```scala
object Foo extends App {
  println(42)
}
```

Save the file and run it via the Command Palette or the inline `run` widget. All is fine, `42` is printed.

Now change `42` to `43` and, without explicitly saving, run the file again. It _looks_ the updated code is respected, because the circle in the file's tab that indicates unsaved changes has vanished before execution. But still `42` is printed. Run it again, and things work fine again.

### Expected behavior

Latest code is used during execution.

### Operating system

macOS

### Editor/Extension

VS Code

### Version of Metals

v0.11.12

**Comment by tgodzik** [comment]
> Now change 42 to 43 and, without explicitly saving, run the file again.

I think this might actually be a bug. How did the file get saved actually? Was it done automatically? I think there might be an issue when we use the Task View in VS Code, where it might run the old code.

**Comment by izzyreal** [comment]
@tgodzik Yes, it gets saved automatically when you perform the `run` action.

**Comment by ckipp01** [comment]
> @tgodzik Yes, it gets saved automatically when you perform the run action.

Wait I think I'm confused, because you said

> But still 42 is printed.

Meaning that I _did't_ think the file was actually saved. For example in nvim-metals I just tested this and indeed the file won't be saved, so it makes sense that 42 is still printed _until_ the save happens. This is probably VS Code specific then?

**Comment by tgodzik** [comment]
Though, we don't save the file automatically, only when running via DAP :thinking:

**Comment by izzyreal** [comment]
@ckipp01 It may well be VS Code specific, or even unrelated to Metals.

From a visual perspective, the impression is that the file gets saved before execution. But it's not clear whether that is what happens under the hood, and whether the file also gets compiled before it gets executed.

**Comment by tgodzik** [comment]
Ok, no, actually it looks like it get saved in VS Code, not sure why :thinking:  Will look into that!

**Comment by tgodzik** [comment]
Looked into it and it seems that we don't properly wait for the save, but not sure exactly why since even adding it explicitly doesn't solve it. We need to get back to it later
